subsys/bluetooth/mesh/access.c

```diff
@@ -147,7 +147,25 @@ static void publish_sent(int err, void *user_data)
 	}
 }
 
+static void publish_start(u16_t duration, int err, void *user_data)
+{
+	struct bt_mesh_model *mod = user_data;
+
+	if (err) {
+		BT_ERR("Failed to publish: err %d", err);
+		return;
+	}
+
+	/* pub->count == 0 means we're at the beginning of a new periood.
+	 * Initialize the timestamp for this.
+	 */
+	if (!mod->pub->count) {
+		mod->pub->period_start = k_uptime_get_32();
+	}
```

Comment:
> pub->count == 0 means we're at the beginning of a new periood.

Is this right? Wouldn't it be pub_transmit_count for the `count > 0` scenario, and 0 on the last packet in the publish?

Reply:
@trond-snekvik ugh.. I think you're right. The beginning of the period should be the beginning of the first transmission, not the last one.

Reply:
@trond-snekvik so we should be initialising period_start when count is equal to the configured retransmission count, right?

Reply:
@trond-snekvik updated the PR, could you take another look?

Reply:
This looks good to me. I like it much better than the initial version too, it never leaves period_start uninitialized 👍